### DISCUSSION TITLE
Egeria connector

### DISCUSSION THREAD

**Discussion by smotornyuk** [ckan/ckan#6080] [Ideas]
As with any project, CKAN has a lot of bright sides and a few problems:) One representative of the latter group is data exchange between applications. There are three main tools available right now for this purpose, but all of them are CKANish and sometimes lack of abstraction prevents us from reaching farther places:

1. API - for taking data from CKAN by 3rd-party applications 
2. ckanext-harvest - for bringing data from anywhere into CKAN
3. ckanext-syndicate - for pushing data from CKAN to anywhere

There is a tool designed exactly for this case, which standardizes the way of data exchange between platforms and can provide CKAN with abilities both to share data from other applications and take it from them. The one, mentioned in the title,  [Egeria](https://egeria.odpi.org/)

In simple terms, it just forms data-flow veins and CKAN(as well as any other application) can be connected to these veins and either inject its datasets into this **body** or pick some details from data-flow. The only two things we need to join this system is [Open Metadata Repository Connector](https://wiki.lfaidata.foundation/display/EG/Implement+an+Open+Metadata+Repository+Connector) and a lot of enthusiasm

**Comment by anuveyatsu** [comment]
@smotornyuk thanks for sharing this 👍 I think [DCAT extension](https://github.com/ckan/ckanext-dcat) is also for this purpose (not data but metadata).

**Reply by smotornyuk** [reply]
Yep, exactly. But there is one extra bit, that I like about egeria. While DCAT and other standards dictate a way of metadata representation, data discovery itself is out of their control. And every tool similar to CKAN(Socrata, ArcGis, etc) has its own set of API, that one must study in order to get more data into the portal(CKAN has a number of harvesters, but if there will be yet another platform, we have to create yet another harvester) and syndicate data to other portals(even if other platforms understand DCAT, you still have to find out **where** and **how** metadata must be posted). And for every new tool, it will be even harder because this new tool must somehow create connectors for every single existing platform before it can use external data. Of course, I like CKAN, but we can't force everyone to use it. So we have to deal with a constantly changing variety of applications and tools, every of which has a unique way of dealing with metadata :(

What egeria suggests is more like messaging protocol. Whenever you have something to share, you just send metadata down through the pipe and every interested party can somehow use this data. You can subscribe to notifications and take interesting items from the global data flow. And, theoretically, you can search for data inside this data flow if you don't have anything relevant available locally.

**Comment by ccancellieri** [comment]
Is this something like a client 'watching' a metadata and (optional: check for 'activities' and if something relevant has changed) then the clients (watcher) will pull the changes? 
So it can be done as pull  from clients 'on receiving alerts' from 'watched' metadata.

**Comment by jqnatividad** [comment]
Perhaps we can do both...

IRL, discovery, especially for public sites, means being indexed by Google.

Maybe we should make DCAT a standard, built-in plugin as it supports schema.org markup and is [supported by Google Dataset Search](https://ckan.org/blog/make-open-data-discoverable-for-search-engines) (GDS), especially now that [GDS is out of beta](https://blog.google/products/search/discovering-millions-datasets-web).

For finer-grained, more transactional metadata discovery and support of Data Exchange use cases where CKAN is used as an enterprise data catalog (beyond just publishing open data), it appears Egeria is better suited than DCAT.
